### PR TITLE
credential test: Refactor

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
credential test file has a lot of duplicated code. 
The tests can be merged together with describe table.

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/compute/credentials.go [hunk 1]
```diff
@@ -53,82 +53,91 @@ var _ = SIGDescribe("Guest Access Credentials", func() {
...
+	keysSecretData := libsecret.DataBytes{
+		"my-key1": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"),
+		"my-key2": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"),
+		"my-key3": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"),
+	}
 
-		It("[test_id:6220]should propagate public ssh keys", func() {
-			const secretID = "my-pub-key"
-			vmi := libvmifact.NewFedora(withSSHPK(secretID, withQuestAgentPropagationMethod))
+	userData := fmt.Sprintf(
```
Comment (0xFelix) [inline tests/compute/credentials.go:66]: Can you make this a const and remove the part setting the password? In test case `using qemu guest agent` you are not setting the password and still it is possible to login in the Expecter before (Password `fedora` is already the default?)

File: tests/compute/credentials.go [hunk 2]
```diff
@@ -276,3 +216,21 @@ func withSSHPK(secretName string, propagationMethod v1.SSHPublicKeyAccessCredent
 		}
 	}
 }
+
+func withPassword(secretName string) libvmi.Option {
```
Comment (0xFelix) [inline tests/compute/credentials.go:220]: We might have a `libvmi` Options for this and `withSSHPK` soon.

File: tests/compute/credentials.go [hunk 3]
```diff
@@ -53,82 +53,91 @@ var _ = SIGDescribe("Guest Access Credentials", func() {
...
-			vmi = libvmops.RunVMIAndExpectLaunch(vmi, fedoraRunningTimeout)
+		vmi := libvmifact.NewFedora(options...)
+		vmi = libvmops.RunVMIAndExpectLaunch(vmi, fedoraRunningTimeout)
 
-			By("Waiting for agent to connect")
-			Eventually(matcher.ThisVMI(vmi), guestAgentConnetTimeout, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+		By("Waiting for agent to connect")
+		Eventually(matcher.ThisVMI(vmi), guestAgentConnetTimeout, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
 
+		if withAccessCredential {
```
Comment (0xFelix) [inline tests/compute/credentials.go:81]: All test cases in this Table are using the Access Credentials API. You should drop this bool and check.

File: tests/compute/credentials.go
Comment (0xFelix) [inline tests/compute/credentials.go:1]: I'd drop the comment in L148

Comment (0xFelix) [inline tests/compute/credentials.go:1]: typo: `guestAgentConnetTimeout`

Comment (0xFelix) [inline tests/compute/credentials.go:1]: Can you dot-import `kubevirt.io/kubevirt/tests/framework/matcher` to make using the matchers look nicer?

File: tests/compute/credentials.go [hunk 4]
```diff
@@ -53,82 +53,91 @@ var _ = SIGDescribe("Guest Access Credentials", func() {
...
+		fedoraPassword,
+	)
 
-			By("Creating a secret with three ssh keys")
-			createNewSecret(testsuite.GetTestNamespace(vmi), secretID, map[string][]byte{
-				"my-key1": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"),
-				"my-key2": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"),
-				"my-key3": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"),
-			})
+	DescribeTable("should have ssh-key under authorized keys added", func(secretID string, secretData libsecret.DataBytes, withAccessCredential bool, options ...libvmi.Option) {
```
Comment (0xFelix) [inline tests/compute/credentials.go:71]: All test cases use the same value for `secretID` and `secretData`. Can you drop the extra table args and use the constants directly instead?

File: tests/compute/credentials.go (hunk 3 above)
Comment (fossedihelm) [reply tests/compute/credentials.go]: AccessCredential condition is populated only with `QemuGuestAgentUserPasswordAccessCredentialPropagation`

**Comment by fossedihelm** [comment]
@0xFelix PTAL

File: tests/compute/credentials.go (hunk 2 above)
Comment (0xFelix) [reply tests/compute/credentials.go:214]: https://github.com/kubevirt/kubevirt/pull/12791

It was merged. The `libvmi` options are called `WithAccessCredentialSSHPublicKey` and `WithAccessCredentialUserPassword`.

File: tests/compute/credentials.go
Comment (0xFelix) [reply tests/compute/credentials.go:1]: In general, yes. For ginkgo/gomega we dot import. Not sure about other matchers to be used with ginkgo/gomega.

Comment (fossedihelm) [reply tests/compute/credentials.go:1]: From my point of view we should not use dot imports. We can have an exception for ginkgo/gomega but will not extend this to all the "match" related stuff

**Review by 0xFelix — COMMENTED** [review]
/approve

Thanks

**Review by jean-edouard — APPROVED** [review]

**Pull request merged**